realm/src/androidTest/java/io/realm/RealmCursorTest.java

```diff
@@ -0,0 +1,595 @@
+/*
+ * Copyright 2015 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm;
+
+import android.database.CharArrayBuffer;
+import android.database.ContentObserver;
+import android.database.Cursor;
+import android.database.DataSetObserver;
+import android.os.Bundle;
+import android.os.Handler;
+import android.test.AndroidTestCase;
+
+import java.util.Date;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import io.realm.android.RealmCursor;
+import io.realm.entities.AllTypes;
+import io.realm.entities.AnnotationNameConventions;
+import io.realm.entities.Dog;
+
+import static io.realm.internal.test.ExtraTests.assertArrayEquals;
+
+public class RealmCursorTest extends AndroidTestCase {
+
+    private static final int SIZE = 10;
+    public static final int NOT_FOUND = -1;
+
+    private Realm realm;
+    private RealmCursor cursor;
+
+    private enum CursorGetter {
+        STRING, SHORT, INT, LONG, FLOAT, DOUBLE, BLOB;
+    }
+
+    @Override
+    protected void setUp() throws Exception {
+        super.setUp();
+        Realm.deleteRealmFile(getContext());
+        realm = Realm.getInstance(getContext());
+        populateTestRealm(realm, SIZE);
+        cursor = realm.allObjects(AllTypes.class).getCursor();
+    }
+
+    private void populateTestRealm(Realm realm, int objects) {
+        realm.beginTransaction();
+        for (int i = 0; i < objects; i++) {
+            AllTypes allTypes = realm.createObject(AllTypes.class);
+            allTypes.setColumnBoolean((i % 3) == 0);
+            allTypes.setColumnBinary(new byte[]{1, 2, 3});
+            allTypes.setColumnDate(new Date());
+            allTypes.setColumnDouble(3.1415);
+            allTypes.setColumnFloat(1.234567f + i);
+            allTypes.setColumnString("test data " + i);
+            allTypes.setColumnLong(i);
+            Dog dog = realm.createObject(Dog.class);
+            dog.setName("Foo " + i);
+            allTypes.setColumnRealmObject(dog);
+        }
+        realm.commitTransaction();
+    }
+
+    @Override
+    protected void tearDown() throws Exception {
+        super.tearDown();
```

Comment:
Nope, it is perfectly safe to close the Realm after the super class has been called